### PR TITLE
Bug 1209402 - [TaskManager] Implement New Sheet and New Private Sheet.

### PR DESCRIPTION
**Pull request by mcav**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=be2ede1a7b801db103cd4c0829af4d271a4c871e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=84426d83c5eb91fd953a32995f2897fba3e94ae5

File: apps/system/js/task_manager.js [hunk 1]
```diff
@@ -538,9 +553,38 @@ TaskManager.prototype = {
...
+   */
+  openNewSheet({ isPrivate } = {}) {
+    var appWindow;
+
+    if (isPrivate) {
+      appWindow = new AppWindow(new BrowserConfigHelper({
+        manifestURL: 'app://search.gaiamobile.org/manifest.webapp',
+        url: 'app://search.gaiamobile.org/newtab.html?private=1',
+        isPrivate: true
+      }));
```
Comment (etiennesegonzac) [inline apps/system/js/task_manager.js]: looks like this is not actually opening a private window :)
(the private window have a different visual treatment for chrome and in the card view, you can use the "..." menu in the browser chrome to open one and see the different).

Haven't dug into the details but https://github.com/mozilla-b2g/gaia/blob/49a75c53b3119cf73f1dcec2926fe4a8ebde9da9/apps/system/js/browser.js#L59-L70 works

File: apps/system/test/marionette/task_manager_test.js [hunk 2]
```diff
@@ -179,6 +179,23 @@ marionette('Task Manager', function() {
...
+    test('should open a new sheet', function() {
+      taskManager.newSheetButton.tap();
+      system.waitForBrowser('app://search.gaiamobile.org/newtab.html');
+    });
+
+    test('should open a new private sheet', function() {
+      taskManager.newPrivateSheetButton.tap();
+      system.waitForBrowser(
+        'app://search.gaiamobile.org/newtab.html?private=1');
+    });
```
Comment (etiennesegonzac) [inline apps/system/test/marionette/task_manager_test.js]: we should assert that the appWindow has the `.private` class here.
It's not perfect but it'll be inline with the other private browsing tests.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=64b2ad0296dd7fcb57bf73197d92429c6141f2df

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=1e98c3c08b82e1f676b1b35a2fabd3eb33c79a60

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d4b349a36ef8be4446856d7fb5a5eea70c819f25

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=91b32b7378cdefd205f792576ae74b336efb64fa

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=aa52cea6078961475a0797d487eac0b2e046cd6e

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=ea68151257a69e105ad3281a3400165215b64ab0

**Pull request merged**